examples/crowdfunding/test/failing-tests.js

```diff
@@ -0,0 +1,437 @@
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: donor.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), 'RUNTIME_ERR1008: Index out of bound');
+  });
+
+  it('should fail if transaction is signed by wrong lsig', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    const escrowProg = getProgram('wrongEscrow.teal', { APP_ID: applicationId });
+    const wrongLsig = runtime.getLogicSig(escrowProg, []);
+    runtime.setRoundAndTimestamp(5, endDate.getTime() + 11);
+    appArgs = [stringToBytes('claim')];
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: creator.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        lsig: wrongLsig,
+        payFlags: { totalFee: 1000, closeRemainderTo: creator.address }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), 'RUNTIME_ERR1301: logic signature validation failed.');
+  });
+
+  it('should fail if transaction is lsig is missing', () => {
+    updateAndOptIn();
+    runtime.executeTx(donateTxGroup);
+    runtime.executeTx(donateTxGroup);
+    runtime.setRoundAndTimestamp(5, endDate.getTime() + 11);
+    appArgs = [stringToBytes('claim')];
+    const txGroup = [
+      {
+        type: types.TransactionType.CallNoOpSSC,
+        sign: types.SignType.SecretKey,
+        fromAccount: creator.account,
+        appId: applicationId,
+        payFlags: { totalFee: 1000 },
+        appArgs: appArgs
+      },
+      {
+        type: types.TransactionType.TransferAlgo,
+        sign: types.SignType.LogicSignature,
+        fromAccount: escrow.account,
+        toAccountAddr: creator.address,
+        amountMicroAlgos: 0,
+        payFlags: { totalFee: 1000, closeRemainderTo: creator.address }
+      }
+    ];
+
+    assert.throws(() => runtime.executeTx(txGroup), 'RUNTIME_ERR1300: logic signature not found');
+  });
+
+  it('should fail to delete app because escrow account balance is not empty', () => {
+    updateAndOptIn();
+    runtime.setRoundAndTimestamp(5, fundCloseDate.getTime() + 12);
+    const deleteTx = {
+      type: types.TransactionType.DeleteSSC,
+      sign: types.SignType.SecretKey,
+      fromAccount: creator.account,
+      appId: applicationId,
+      payFlags: { totalFee: 1000 },
+      appArgs: [],
+      accounts: [escrow.address] //  AppAccounts
+    };
+
+    assert.throws(() => runtime.executeTx(deleteTx), 'RUNTIME_ERR1008: Index out of bound');
```

Comment:
Why `Index out of bound`? Shouldn't the transaction be rejected? Can you please confirm?

Reply:
the way logic is written it will throw index out of bound, the other way would be to change crowdfunding.teal
but i think we shouldn't because it is same in tutorial

Reply:
Ok, agreed. Just wanted to make sure it's not a bug.